### DISCUSSION TITLE
出个教程如何在调用链入口处生成 TraceID 吧

### DISCUSSION THREAD

**Discussion by StLeoX** [deepflowio/deepflow#5254] [General]
现在 DeepFlow 无法生成 TraceID，指导一下生产环境中用什么组件能在调用链入口处生成 TraceID 吧。

**Comment by sharang** [comment]
DeepFlow 不生成 TraceID，也不传播 TraceID。

DeepFlow 使用 eBPF 自动计算 syscall_trace_id 用于分布式追踪，不需要依赖一个全局的 TraceID。

但在有一些跨线程的场景下，DeepFlow 需要依赖 APM Agent 向 Protocol Header 中注入的 TraceID（DeepFlow 会自动解析他们）。

**Comment by sharang** [comment]
另外，也可以考虑在网关上开启 X-Request-ID，https://deepflow.io/docs/zh/features/distributed-tracing/auto-tracing/#x-request-id

使用它可以作为入口的 TraceID
